qa/tasks/cephfs/test_client_recovery.py

```diff
@@ -567,3 +567,33 @@ def test_abort_conn(self):
         self.assert_session_state(gid, "open")
         time.sleep(session_timeout * 1.5)  # Long enough for MDS to consider session stale
         self.assert_session_state(gid, "stale")
+
+    def test_dont_mark_unresponsive_client_stale(self):
+        """
+        Test that an unresponsive client holding no caps is evicted directly,
+        without being marked as stale even after session_timeout, at
+        session_autoclose.
+
+        To conduct this test we need more that two clients since a single
+        client is never evcited by MDS.
+        """
+        SESSION_TIMEOUT = 30
+        SESSION_AUTOCLOSE = 50
+        mount_a_gid = self.mount_a.get_global_id()
+        mount_a_pid = self.mount_a.get_client_pid()
+        self.fs.set_var('session_timeout', SESSION_TIMEOUT)
+        self.fs.set_var('session_autoclose', SESSION_AUTOCLOSE)
+        self.assert_session_count(2, self.fs.mds_asok(['session', 'ls']))
+
+        self.mount_a.run_shell(['mkdir', 'dir'])
+        self.mount_a.send_signal('sigstop')
+        time.sleep(SESSION_TIMEOUT + 2)
+        self.assert_session_state(mount_a_gid, "open")
+
+        # actual test part
+        self.mount_b.run_shell(['stat', 'dir'])
+        time.sleep(SESSION_AUTOCLOSE - SESSION_TIMEOUT + 2)
+        self.assert_session_count(1, self.fs.mds_asok(['session', 'ls']))
+
+        # Kill the corresponding ceph-fuse process
+        self.mount_b.run_shell(['kill', '-sigkill', mount_a_pid])
```

Comment:
Why do we want to kill mount_b?

Reply:
I don't recall why I wrote it but the test hangs without it.

Reply:
The client should be better about detecting blacklisting. Try again?

Reply:
> Why do we want to kill mount_b?

By the way I am killing `mount_a` from `mount_b.run_shell`.

> The client should be better about detecting blacklisting. Try again?

I don't understand what you are trying to say here.